app/code/Magento/Developer/Model/XmlCatalog/Format/VsCode.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Developer\Model\XmlCatalog\Format;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\DomDocument\DomDocumentFactory;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Filesystem\Directory\ReadFactory;
+use Magento\Framework\Filesystem\Directory\ReadInterface;
+use Magento\Framework\Filesystem\File\WriteFactory;
+
+/**
+ * Class VsCode generates URN catalog for VsCode
+ */
+class VsCode implements FormatInterface
+{
+    private const PROJECT_PATH_IDENTIFIER = '..';
+
+    /**
+     * @var ReadInterface
+     */
+    private $currentDirRead;
+
+    /**
+     * @var WriteFactory
+     */
+    private $fileWriteFactory;
+
+    /**
+     * @var DomDocumentFactory
+     */
+    private $domDocumentFactory;
+
+    /**
+     * @param ReadFactory $readFactory
+     * @param WriteFactory $fileWriteFactory
+     * @param DomDocumentFactory $domDocumentFactory
+     */
+    public function __construct(
+        ReadFactory $readFactory,
+        WriteFactory $fileWriteFactory,
+        DomDocumentFactory $domDocumentFactory = null
+    ) {
+        $this->currentDirRead = $readFactory->create(getcwd());
+        $this->fileWriteFactory = $fileWriteFactory;
+        $this->domDocumentFactory = $domDocumentFactory ?: ObjectManager::getInstance()->get(DomDocumentFactory::class);
```

Comment:
When creating a completely new class - you don't need to provide backward compatible class loader.